corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -407,6 +437,40 @@ def delete_all_cases(domain=None):
             cursor.execute('SELECT delete_all_cases(%s)', [domain])
 
     @staticmethod
+    def get_all_cases_modified_since(server_modified_on_since=None, chunk_size=500):
+        start_from = server_modified_on_since or datetime.min
+        # todo: this will greedily query the same cases multiple times in a sharded
+        # setup. We should make it smarter
+        batch = CaseAccessorSQL.get_cases_modified_since(start_from, limit=chunk_size)
+        while len(batch) == chunk_size:
+            for case in batch:
+                yield case
+                next_start_from = case.server_modified_on
+
+            # make sure we are making progress
+            assert next_start_from > start_from
+            start_from = next_start_from
+            batch = CaseAccessorSQL.get_cases_modified_since(start_from, limit=chunk_size)
+
+        # last batch
+        for case in batch:
+            yield case
+
+    @staticmethod
+    def get_cases_modified_since(server_modified_on_since=None, limit=500):
+        """
+        Iterate through all cases in the entire database, optionally modified since
+        a specific date
+        """
+        if server_modified_on_since is None:
+            server_modified_on_since = datetime.min
+        results = list(CommCareCaseSQL.objects.raw('SELECT * FROM get_all_cases_modified_since(%s, %s)',
+                                                [server_modified_on_since, limit]))
```

Comment:
I think the most important criteria is that once we start iterating through dates, nothing can ever get inserted into the data that we've already seen. So from that perspective `server_modified_on` is ok, though could result in some cases being returned multiple times if they are modified while we're iterating. I don't think we can use `opened_on` because that's a phone date which could be in the past. If we had `server_opened_on` that would probably be best (should we consider adding that to the model?)